src/libostree/ostree-repo-static-delta-processing.c

```diff
@@ -333,28 +335,25 @@ _ostree_static_delta_part_execute_async (OstreeRepo      *repo,
   asyncdata->part = g_variant_ref (part);
   asyncdata->cancellable = cancellable ? g_object_ref (cancellable) : NULL;
 
-  asyncdata->result = g_simple_async_result_new ((GObject*) repo,
-                                                 callback, user_data,
-                                                 _ostree_static_delta_part_execute_async);
-
-  g_simple_async_result_set_op_res_gpointer (asyncdata->result, asyncdata,
-                                             static_delta_part_execute_async_data_free);
-  g_simple_async_result_run_in_thread (asyncdata->result, static_delta_part_execute_thread, G_PRIORITY_DEFAULT, cancellable);
-  g_object_unref (asyncdata->result);
+  task = g_task_new (G_OBJECT (repo), cancellable, callback, user_data);
```

Comment:
I think this can now just be `g_autoptr(GTask) task = g_task_new (...)` and then we can drop the manual `g_object_unref()` below.